File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/V086GameEventHandler.java
package org.emulinker.kaillera.controller.v086.action;

import org.emulinker.kaillera.controller.v086.V086Controller.V086ClientHandler;
import org.emulinker.kaillera.model.event.GameEvent;

public interface V086GameEventHandler<T extends GameEvent> {
  @Override
  public String toString();

  public void handleEvent(T event, V086ClientHandler clientHandler);

  public int getHandledEventCount();
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/V086GameEventHandler.kt
package org.emulinker.kaillera.controller.v086.action

import org.emulinker.kaillera.controller.v086.V086ClientHandler
import org.emulinker.kaillera.model.event.GameEvent

interface V086GameEventHandler<in T : GameEvent?> {
  @Deprecated("Structure this in a different way")
  override fun toString(): String

  fun handleEvent(event: T, clientHandler: V086ClientHandler)
  val handledEventCount: Int
}
